packages/asset-swapper/src/utils/aggregate.ts

```diff
@@ -0,0 +1,780 @@
+        }
+    }
+    return orders;
+}
+
+// Convert buy fills into orders.
+function createBuyOrdersFromPath(
+    inputToken: string,
+    outputToken: string,
+    path: Fill[],
+    bridgeSlippage: number,
+): OrderWithoutDomain[] {
+    const orders: OrderWithoutDomain[] = [];
+    for (const fill of path) {
+        const source = (fill.data as FillData).source;
+        if (source === ERC20BridgeSource.Native) {
+            orders.push((fill.data as NativeFillData).order);
+        } else if (source === ERC20BridgeSource.Eth2Dai) {
+            orders.push(createEth2DaiOrder(inputToken, outputToken, fill.input, fill.output, bridgeSlippage, true));
+        } else if (source === ERC20BridgeSource.Uniswap) {
+            orders.push(createUniswapOrder(inputToken, outputToken, fill.input, fill.output, bridgeSlippage, true));
+        } else {
+            throw new Error(`invalid buy fill source: ${source}`);
+        }
+    }
+    return orders;
+}
+
+function createKyberOrder(
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return createBridgeOrder(
+        KYBER_BRIDGE_ADDRESS,
+        makerToken,
+        takerToken,
+        makerAssetAmount,
+        takerAssetAmount,
+        slippage,
+        isBuy,
+    );
+}
+
+function createEth2DaiOrder(
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return createBridgeOrder(
+        ETH2DAI_BRIDGE_ADDRESS,
+        makerToken,
+        takerToken,
+        makerAssetAmount,
+        takerAssetAmount,
+        slippage,
+        isBuy,
+    );
+}
+
+function createUniswapOrder(
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return createBridgeOrder(
+        UNISWAP_BRIDGE_ADDRESS,
+        makerToken,
+        takerToken,
+        makerAssetAmount,
+        takerAssetAmount,
+        slippage,
+        isBuy,
+    );
+}
+
+function createBridgeOrder(
+    bridgeAddress: string,
+    makerToken: string,
+    takerToken: string,
+    makerAssetAmount: BigNumber,
+    takerAssetAmount: BigNumber,
+    slippage: number,
+    isBuy: boolean = false,
+): OrderWithoutDomain {
+    return {
+        makerAddress: bridgeAddress,
+        makerAssetData: createBridgeAssetData(makerToken, bridgeAddress, createBridgeData(takerToken)),
+        takerAssetData: createERC20AssetData(takerToken),
+        ...createCommonOrderFields(makerAssetAmount, takerAssetAmount, slippage, isBuy),
+    };
+}
+
+/**
+ * Create ERC20Bridge asset data.
+ */
+export function createBridgeAssetData(tokenAddress: string, bridgeAddress: string, bridgeData: string): string {
```

Comment:
We should add an asana task to get this encoding/decoding into `DevUtils` and re-use from that.

Reply:
There's a good chance it won't fit. `DevUtils` needs so much TLC. Like why is `chai` a non-dev dep for `DevUtils`? :woman_shrugging:

Reply:
(if it fits)